### PR TITLE
 Add Feature Inventory screen (internal)

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/1198194956794324/1207827226270692/f

### Description
Adds a Feature Flag Inventory screen in internal settings.

Incidentally, the PR had to fix some tests and took the opportunity to use the `FakeFeatureToggleFactory` rather than mocking the feature flags.
Because of that I also had to add update/add some lint-baselines.xml

### Steps to test this PR
- [ ] install from this branch and launch the app
- [ ] go to Settings -> scroll down -> Feature Flag Inventory
- [ ] verify the list of feature flags appear
- [ ] use the search box to search for features
- [ ] verify filtering is done correctly
- [ ] toggle one of the features ON and/or OFF
- [ ] verify the toggles actually toggle the feature state and whatever code guarded by the feature reacts properly
- [ ] tap on the feature name
- [ ] verify the `Feature Flag Info` bottom sheet pops up and contains the FF information (this info is only present for features that are in the remote config)

### CODE REVIEW

**Comment by aitorvs** [comment]
* **#5001** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5001?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @aitorvs and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by cmonfortep — COMMENTED** [review]
Changes LGTM, but the switches are not working for me when testing. 

Will leave this pending from testing.

File: feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            views.forEach { binding.featureToggle.addView(it) }
+            binding.progress.isVisible = false
+            binding.container.isVisible = true
+        }
+    }
+
+    private suspend fun getFeatureViews(): List<View> = withContext(dispatcherProvider.io()) {
+        val toggles = this@FeatureToggleInventoryActivity.toggles.await()
+        val match = featureNameFilter.get().lowercase()
+        logcat { "aitor $match" }
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt:121]: same

File: feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            .sortedBy { it.featureName().name.lowercase() }
+        val subFeatures = toggles
+            .filter { it.featureName().parentName != null }
+            .sortedBy { it.featureName().name.lowercase() }
+        val features = mutableListOf<Toggle>().apply {
+            for (parent in parentFeature) {
+                add(parent)
+                addAll(subFeatures.filter { it.featureName().parentName == parent.featureName().name })
+            }
+            // Apply search box filter if needed
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt:133]: will do

File: feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            }
+        }
+
+        val views = features.map { feature ->
+            OneLineListItem(this@FeatureToggleInventoryActivity).apply {
+                if (feature.featureName().parentName != null) {
+                    setPrimaryText("\u2514   ${feature.featureName().name}")
+                    setPrimaryTextTruncation(truncated = true)
+                } else {
+                    setPrimaryText(feature.featureName().name)
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply feature-toggles/feature-toggles-internal/src/main/java/com/duckduckgo/examplefeature/internal/ui/FeatureToggleInventoryActivity.kt:148]: lol

File: feature-toggles/feature-toggles-impl/src/main/java/com/duckduckgo/feature/toggles/impl/RealFeatureTogglesInventory.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    override suspend fun getAll(): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext toggles.flatMap { it.getAll() }.distinctBy { it.featureName() }
+    }
+}
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = MainProcessLifecycleObserver::class,
+)
+class FeatureToggleInventoryTest @Inject constructor(
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply feature-toggles/feature-toggles-impl/src/main/java/com/duckduckgo/feature/toggles/impl/RealFeatureTogglesInventory.kt:49]: ah yeah, left over from some testing

File: feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureTogglesInventory.kt [hunk 5]
```diff
@@ -14,14 +14,11 @@
...
-
-class FakeCredentialsSyncLocalValidationFeature : CredentialsSyncLocalValidationFeature {
-    var enabled = true
-
-    override fun self(): Toggle = TestToggle(enabled)
+interface FeatureTogglesInventory {
+    /**
+     * @return returns a list of ALL feature flags (aka [Toggle]s) currently used in the app
+     */
+    suspend fun getAll(): List<Toggle>
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureTogglesInventory.kt:23]: Yeah, we'll use it soon in another place

**Review by cmonfortep — APPROVED** [review]
LGTM 👍 🚢 it

**Pull request merged**